GraficaEngine/GraficaEngine/Scripts/ThirdPersonCameraController.cpp

```diff
@@ -19,15 +19,15 @@ ThirdPersonCameraController *ThirdPersonCameraController::clone() const
 
 void ThirdPersonCameraController::update()
 {
-    Engine::Transform &transform = gameObject->transform;
-    Engine::BaseGameObject *player = gameObject->getScene()->getGameObjectWithTag("player");
+    auto& transform = gameObject->transform;
+    auto* player = gameObject->getScene()->getGameObjectWithTag("player");
 
     if (player == nullptr) {
         return;
     }
 
-    Engine::Transform &playerTransform = player->transform;
-    Engine::Input &input = Engine::Input::getInstance();
+    auto& playerTransform = player->transform;
+    auto& input = Engine::Input::getInstance();
```

Comment:
Hmm. I am not 100% sure this creates the correct reference. I remember a warning from VSCode specifying that auto does not work well with references. But maybe adding the explicit & fixes it?

Reply:
The script apparently works